demo/Demo/Menus.elm

```diff
@@ -0,0 +1,222 @@
+  Signal.Address Menu.Action -> Menu.Model -> List Style -> List Html -> Html
+
+
+type alias View' =
+  Signal.Address Menu.Action -> Menu.Model -> Html
+
+
+view' : View -> List Style -> Html -> Signal.Address Menu.Action -> Menu.Model -> Html
+view' view coloring elem addr model =
+  view addr model coloring [elem]
+
+
+menus : List (Index, (Style, String))
+menus =
+  [ (Menu.bottomLeft, "Lower left")
+  , (Menu.bottomRight, "Lower right")
+  , (Menu.topLeft, "Top left")
+  , (Menu.topRight, "Top right")
+  ]
+  |> List.indexedMap (\i r -> (i, r))
+
+
+model : Model
+model =
+  { menus = menus
+         |> List.map (\(i, (align,_)) -> (i, Menu.model True align))
+         |> Dict.fromList
+  }
+
+
+-- ACTION, UPDATE
+
+
+type Action
+  = Action Index Menu.Action
+
+
+type alias Model =
+  { menus : Dict.Dict Index Menu.Model
+  }
+
+
+update : Action -> Model -> (Model, Effects.Effects Action)
+update action model =
+  case action of
+    Action idx action ->
+      Dict.get idx model.menus
+      |> Maybe.map (\m0 ->
+        let
+          (m1, e) = Menu.update action m0
+        in
+          ({ model | menus = Dict.insert idx  m1 model.menus }, Effects.map (Action idx) e)
+      )
+      |> Maybe.withDefault (model, Effects.none)
+
+
+-- VIEW
+
+
+view : Signal.Address Action -> Model -> Html
+view addr model =
+
+  menus
+  |> List.map (\( idx, (menu, description) ) ->
+       let
+         model' = model.menus
+                |> Dict.get idx
+                |> Maybe.withDefault (Menu.model True Menu.unaligned)
+         items  =
+           [ Menu.item False True  (text "Some Action")
+           , Menu.item True  True  (text "Another Action")
+           , Menu.item False False (text "Disabled Action")
+           , Menu.item False True  (text "Yet Another Action")
+           ]
+       in
+         Grid.cell
+         [ Grid.size Grid.All 6
+         ]
+         [ container addr description idx model' items
+         ]
+     )
+  |> Grid.grid []
+  |> flip (::) []
+  |> Page.body2 "Menus" srcUrl intro references
+
+
+container : Signal.Address Action -> String -> Index -> Menu.Model -> List Menu.Item -> Html
+container addr description idx model' items =
+
+  let
+    bar rightAlign =
+      let
+        align =
+          if rightAlign then ("right", "16px") else ("left", "16px")
+      in
+        div
+          [ class "bar"
+          , style
+            [ ("box-sizing", "border-box")
+            , ("background", "#3F51B5")
+            , ("color", "white")
+            , ("width", "100%")
+            , ("padding", "16px")
+            , ("position", "relative")
+            , ("height", "64px")
+            ]
+          ]
+          [ div
+              [ class "wrapper"
+              , style
+                [ ("position", "absolute")
+                , align
+                , ("box-sizing", "border-box")
+                ]
+              ]
+              ( Menu.view (Signal.forwardTo addr (Action idx)) model' items
```

Comment:
Very good! It's great that `items` is a parameter to view and not part of the model.